presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -302,6 +302,18 @@ public void createPartitionedTable(Session session, HiveStorageFormat storageFor
 
         assertUpdate(session, "INSERT INTO test_partitioned_table " + select, 1);
         assertQuery(session, "SELECT * from test_partitioned_table", select);
+        assertQuery(session,
+                "SELECT * from test_partitioned_table WHERE" +
+                        " 'foo' = _partition_string" +
+                        " AND 'bar' = _partition_varchar" +
+                        " AND CAST('boo' AS CHAR(10)) = _partition_char" +
+                        " AND CAST(1 AS TINYINT) = _partition_tinyint" +
+                        " AND CAST(1 AS SMALLINT) = _partition_smallint" +
+                        " AND 1 = _partition_integer" +
+                        " AND CAST(1 AS BIGINT) = _partition_bigint" +
+                        " AND CAST('3.14' AS DECIMAL(3,2)) = _partition_decimal_short" +
```

Comment:
oh, assumeCanonicalPartitionKeys has always been false when I've run this, so it just puts in the wildcard.  I bet we'd have the same problem with short decimal if assumeCanonicalPartitionKeys were true.  

Also, do you know if decimal has the same problem with 0s that all the number/boolean values have?  Do In need to add a check about assumeCanonicalPartitionKeys there too?